InvenTree/build/serializers.py

```diff
@@ -519,6 +520,24 @@ def validate(self, data):
 
         outputs = data.get('outputs', [])
 
+        if common.settings.prevent_build_output_complete_on_incompleted_tests:
+            error_string = ''
+            fail_count = 0
+
+            for output in outputs:
+                stock_item = output['output']
+                if stock_item.hasRequiredTests() and not stock_item.passedAllRequiredTests():
+                    if stock_item.part.trackable:
+                        if len(error_string):
+                            error_string += "<br>"
```

Comment:
Do not pass html through on the error string - this will be provided to the front-end via the API, and the front-end will decide how to render the error(s).

If more than one error string is required, provide as a list of strings,